Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -190,7 +199,53 @@ public class NIOSSLHandler : ChannelInboundHandler, ChannelOutboundHandler, Remo
         self.doFlushReadData(context: context, receiveBuffer: receiveBuffer, readOnEmptyBuffer: true)
         self.writeDataToNetwork(context: context, promise: nil)
     }
-    
+
+    // TODO: how do I test this?
+    public func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+        switch event {
+        case ChannelEvent.inputClosed:
+            userInboundInputClosedTriggered(context: context)
+        default:
+            context.fireUserInboundEventTriggered(event)
+            // Trigger closure of TCP input as this event has not occured although we are in state .inputClosed already.
+            if case .inputClosed = state {
+                context.fireUserInboundEventTriggered(ChannelEvent.inputClosed)
+            }
+        }
+    }
+
+    private func userInboundInputClosedTriggered(context: ChannelHandlerContext) {
+        let channelError: NIOSSLError
+        switch self.state {
+        case .closed, .idle:
+            return
+        case .inputClosed:
+            context.fireUserInboundEventTriggered(ChannelEvent.inputClosed)
```

Comment:
Ideally this isn't right. We should synthesise this event when we _enter_ the `.inputClosed` state, and then suppress it here.